util/postprocess-makedepend.pl

```diff
@@ -0,0 +1,81 @@
+#! /usr/bin/env perl
+# Copyright 2018 The OpenSSL Project Authors. All Rights Reserved.
+#
+# Licensed under the OpenSSL license (the "License").  You may not use
+# this file except in compliance with the License.  You can obtain a copy
+# in the file LICENSE in the source distribution or at
+# https://www.openssl.org/source/license.html
+
+my $producer = $ARGV[0];
+
+while (<STDIN>) {
+    if ($producer eq 'makedepend') {
+
+        # makedepend, in its infinite wisdom, wants to have the object file in
+        # the same directory as the source file.  This doesn't work too well
+        # with out-of-source-tree builds, so we must resort to tricks to get
+        # things right.  The trick is to call makedepend with an extra suffix
+        # that contains the desired object file path, like this:
+        #
+        #     makedepend -f- -o"|dir/foo.o" -- $(CFLAGS) -- ../somewhere/foo.c
+        #
+        # The result will look something like this:
+        #
+        #     ../somewhere/foo|dir/foo.o: deps...
+        #
+        # Which is easy to massage by removing everything up to the first |
+
+        # Remove everything up to the first |
+        s/^.*\|//;
+        # Also, remove any dependency that starts with a /, because those are
+        # typically system headers
+        s/ \/(\\.|[^ ])*//;
+        # Finally, remove all empty lines or comment lines
+        $_ = undef if (/: *$/ || /^(#.*| *)$/);
+        $_.="\n" unless !defined($_) or /\R$/g;
+
+    } elsif ($producer eq 'VMS C') {
+
+        # current versions of DEC / Compaq / HP / VSI C strips away all
+        # directory information from the object file, so we must insert it
+        # back. Just to be safe against future changes, we check that there
+        # really is no directory information.
+
+        my $directory = $ARGV[1];
+
+        # The pattern for target and dependencies will always take this form:
+        #
+        #     target SPACE : SPACE deps
+        #
+        # This is so a volume delimiter (a : without any spaces around it)
+        # won't get mixed up with the target / deps delimiter.  We use this
+        # fact in the regexp below to make sure we do look at the target.
+        s/^/$directory/ unless /^\S+[:<\]]\S+\s+:/;
+
+    } elsif ($producer eq 'Windows C') {
+
+        # For the moment, we only know of one native Windows C compiler, and
```

Comment:
This is not true. As far as actual *knowing* goes you can't deny that you, Richard, know of Intel compiler for Windows :-) :-) :-) What I'm saying is wouldn't it be more appropriate to say that we *support* only one compiler. And it would be *more* than appropriate to not refer to it as "Windows C", but by its real name, Visual C, or MSVC, or VC. Yes, one can argue that just like gcc sets the tone in Unix, VC does the same on Windows, but I'd still be against referring to it as "Windows C". Basically it reads as if there is Windows C *language*, and there is not such thing. Yes, "VMS C" has the same problem. In other words these would be explicit reference to tool-chains. This is actually good example of what I talked about last time in context of template neutrality. My line of reasoning was that one should put weight on tool-chain-specific rules, as opposite to producing infinite amount of parameters. That time I called out at manifest embedding, which could be implemented rather as VC-specific post-link rule addendum, than assuming that every tool-chain has equivalent replacement. Because it's not actually given that every tool-chain even has a post-link step. Same can be said about this dependency thing... Well, as you should know, I'm not on good terms with words, so all this might appear as meaningless rambling... But I do have couple of more concrete remarks, coming next. But here I do suggest to make arguments explicitly about tool-chains.

Reply:
Actually, "VMS C" is as valid as anything, it's what many people are commonly calling what once was ~~DEC~~ / ~~Compaq~~ / HP / VSI C these days (see the problem here? ;-)  some jokingly talked about DECompHP C, and the VSI came along...).  Basically, if you say "VMS C" to VMS folks, they know *exactly* what you're talking about.

But ok, I can go for VC, considering we have some hard-coded VC things.

Side note re this:
> My line of reasoning was that one should put weight on tool-chain-specific rules, as opposite to producing infinite amount of parameters.

I agree with you, and I have some thoughts on how to do that...  but what I have in mind is a *major* change of the config attributes and the build file templates...  To be talked about separately.